core/services/assets/Registry.php

```diff
@@ -449,6 +475,125 @@ public function getCssUrl($namespace, $chunk_name)
+     * @param string $asset_type
+     * @return array
+     * @since $VID:$
+     */
+    private function getDependenciesForAsset($namespace, $chunk_name, $asset_type)
+    {
+        $asset_index = $chunk_name . '.' . $asset_type;
+        if (! isset( $this->dependencies_data[ $namespace ][ $asset_index ])) {
+            $path = isset($this->manifest_data[ $namespace ]['path'])
+                ?
+                $this->manifest_data[ $namespace ]['path']
+                :
+                $this->domain->distributionAssetsPath();
+            $dependencies_index = $chunk_name . '.' . Asset::TYPE_JSON;
+            $file_path = isset($this->manifest_data[ $namespace ][ $dependencies_index ])
+                ? $path . $this->manifest_data[ $namespace ][ $dependencies_index ]
+                :
+                '';
+            $this->dependencies_data[ $namespace ][ $asset_index ] = $file_path !== '' && file_exists($file_path)
+                ? $this->getDependenciesForAssetType($namespace, $asset_type, $file_path, $chunk_name)
+                : [];
+        }
+        return $this->dependencies_data[ $namespace ][ $asset_index ];
+    }
+
+
+    /**
+     * Return dependencies according to asset type.
+     *
+     * For css assets, this filters the auto generated dependencies by css type.
+     *
+     * @param string $namespace
+     * @param string $asset_type
+     * @param string $file_path
+     * @param string $chunk_name
+     * @return array
+     * @since $VID:$
+     */
+    private function getDependenciesForAssetType($namespace, $asset_type, $file_path, $chunk_name)
+    {
+        $asset_dependencies = json_decode(file_get_contents($file_path), true);
+        if ($asset_type === Asset::TYPE_JS) {
+            return $chunk_name === 'eejs-core' ? $asset_dependencies : array_merge(
+                $asset_dependencies,
+                [ CoreAssetManager::JS_HANDLE_JS_CORE ]
+            );
+        }
+        // for css we need to make sure there is actually a css file related to this chunk.
+        if (isset($this->manifest_data[ $namespace ])) {
+            // array of css chunk files for ee.
+            $css_chunks = array_map(
+                function ($value) {
+                    return str_replace('.css', '', $value);
+                },
+                array_filter(
+                    array_keys($this->manifest_data[ $namespace ]),
+                    function ($value) {
+                        return strpos($value, '.css') !== false;
+                    }
+                )
+            );
+            // add known wp chunks with css
+            $css_chunks += [
+                'wp-components',
+                'wp-block-editor',
+                'wp-block-library',
+                'wp-edit-post',
+                'wp-edit-widgets',
+                'wp-editor',
+                'wp-format-library',
+                'wp-list-reusable-blocks',
+                'wp-nux',
```

Comment:
are we certain this will always be the same for ALL CSS assets?

Should this be a static property at the top of the class for greater visibility and easier access if it requires editing?

Reply:
Sure I can make it a static property at the top of the class for easier visibility.  Essentially this is just a map to _check_ against for dependencies.  It _will_ change if wp core has additional css handles.

This whole css dependency generation isn't ideal, the dependency extraction webpack plugin _only_ considers javascript imports.  However, since there is consistency in WP between javascript and css handle names (and we have consistency with EE handles), I'm able to infer (using this hardcoded map) what css dependencies might exist on a bundled file.